File: JavaCore/Q4.java
package JavaCore;
//Q4. Calculate the number & Percentage Of Lowercase Letters,Uppercase Letters, Digits And Other Special Characters In A String
public class Q4 {
    public static void main(String[] args) {
        String s = "hello1@34Up@";
        double l = s.length();

        //Counters:
        Integer LowerCounter = 0;
        Integer UpperCounter = 0;
        Integer DigitCounter = 0;
        Integer SpecialCounter = 0;

        for(int i = 0; i<s.length(); i++){
            char c = s.charAt(i);
            if (Character.isUpperCase(c)){
                UpperCounter++;
            }
            else if(Character.isLowerCase(c)){
                LowerCounter++;
            }
            else if (Character.isDigit(c)){
                DigitCounter++;
            }
            else {
                SpecialCounter++;
            }
        }

        double UpperPercentage = (UpperCounter/l) * 100;
        double LowerPercentage = (LowerCounter/l) * 100;
        double DigitPercentage = (DigitCounter/l) * 100;
        double SpecialPercentage = (SpecialCounter/l) * 100;

        System.out.println("Number of Uppercase Characters: "+ UpperCounter+ "  Percentage: " + UpperPercentage + " %");
        System.out.println("Number of Lowercase Characters: "+ LowerCounter+ "  Percentage: " + LowerPercentage + " %");
        System.out.println("Number of Digits: "+ DigitCounter+"  Percentage: " + DigitPercentage + " %");
        System.out.println("Number of Special characters: "+SpecialCounter+ "  Percentage: "  + SpecialPercentage + " %");

    }
}

File: Kotlin_1/src/exercise1/Q4.kt
package exercise1
//Q4. Calculate the number & Percentage Of Lowercase Letters,Uppercase Letters, Digits And Other Special Characters In A String.
fun main(){
    val input: String = readLine().toString()
    val string = input.trim()
//    val string:String = "hello1@34Up@"
    var upperCounter =0
    var lowerCounter =0
    var digitCounter =0
    var specialCounter =0
    for (i in string){
        when {
            i.isUpperCase() -> {
                upperCounter++
            }
            i.isLowerCase() -> {
                lowerCounter++
            }
            i.isDigit() -> {
                digitCounter++
            }
            else -> {
                specialCounter++
            }
        }

    }

    println("The count is as follows: ")
    println(" Uppercase characters: $upperCounter \n Lowercase characters: $lowerCounter\n Digits: $digitCounter\n Special characters: $specialCounter" )

    val length = string.length.toDouble()
    //println(length)
    val pUpper = (upperCounter/length)*100
    val pLower = (lowerCounter/length)*100
    val pDigit = (digitCounter/length)*100
    val pSpecial = specialCounter/length *100

    println("The percentages are as follows: \nUpper characters: $pUpper \nLower characters: $pLower \nDigits: $pDigit \nSpecial characters: $pSpecial")
}